packages/core/lib/commands/call/run.js

```diff
@@ -0,0 +1,218 @@
+  const { fetchAndCompile } = require("@truffle/fetch-and-compile");
+  const loadConfig = require("../../loadConfig");
+  const DebugUtils = require("@truffle/debug-utils");
+  const web3Utils = require("web3-utils");
+  const { getFirstDefinedValue } = require("../../configAdapter");
+  const { formatBlockSpecifier } = require("../../../../encoder/dist/adapter");
+
+  if (options.url && options.network) {
+    const message =
+      "" +
+      "Mutually exclusive options, --url and --network detected!" +
+      OS.EOL +
+      "Please use either --url or --network and try again." +
+      OS.EOL +
+      "See: https://trufflesuite.com/docs/truffle/reference/truffle-commands/#call" +
+      OS.EOL;
+    throw new TruffleError(message);
+  }
+
+  let config = loadConfig(options);
+  await Environment.detect(config);
+
+  const [contractNameOrAddress, functionNameOrSignature, ...args] = config._;
+  let functionEntry, transaction;
+
+  // Validates the block number and returns the hex string representation of a number or
+  // an error if the string is other than "latest", "pending" and "earliest" or
+  // a hex string representation of a number
+  const blockToFetch = formatBlockSpecifier(config.blockNumber);
+
+  const { encoder, decoder } = config.fetchExternal
+    ? await sourceFromExternal(contractNameOrAddress, config)
+    : await sourceFromLocal(contractNameOrAddress, config);
+
+  try {
+    ({ abi: functionEntry, tx: transaction } = await encoder.encodeTransaction(
+      functionNameOrSignature,
+      args,
+      {
+        allowJson: true,
+        strictBooleans: true
+      }
+    ));
+  } catch (error) {
+    throw new TruffleError(
+      "The function name, function signature or function arguments you entered are invalid!\n" +
+        "Please run the command again with valid function name, function signature and function arguments (if any)!"
+    );
+  }
+
+  if (functionEntry.stateMutability !== "view") {
+    console.log(
+      `WARNING!!! Not a view function\n` +
+        `Any changes this function attempts to make will not be saved\n` +
+        `ABI stateMutability: ${functionEntry.stateMutability}\n`
+    );
+  }
+
+  const provider = new Encoder.ProviderAdapter(config.provider);
+  let decoding;
+
+  // Get the first defined "from" address
+  const fromAddress = getFirstDefinedValue(
+    options.from,
+    config.networks[config.network].from,
+    config.from
+  );
+  checkValidAddress(fromAddress);
+
+  try {
+    const result = await provider.call(
+      fromAddress,
+      transaction.to,
+      transaction.data,
+      blockToFetch
+    );
+
+    [decoding] = await decoder.decodeReturnValue(functionEntry, result, {
+      status: true
+    });
+  } catch (error) {
+    [decoding] = await decoder.decodeReturnValue(functionEntry, error.data, {
+      status: false
+    });
+
+    // Gives a readable interpretation of the panic code returned from the call
+    if (decoding.abi.name === "Panic") {
```

Comment:
This should be outside of the catch block, seems to me.  Also, why does this error case result in a `TruffleError`, but other error cases don't?